### DISCUSSION TITLE
SDK 1.9.0 - Mandatory fields and stars - Best User Experience

### DISCUSSION THREAD

**Discussion by luispozoWS** [OP-TED/eForms-SDK#886] [Q&A] [answered]
Hello,
We are trying to determine which fields should have a star (_*_) to guide buyers on filling mandatory fields and have the esasiest user experience

First, what was the choice for eNotices2 Preview ?
Second, based on fields.json (https://raw.githubusercontent.com/OP-TED/eForms-SDK/1.9.0/fields/fields.json), is it better to :

- Option 1 : Put a star on fields signaled as "mandatory" where their only constraint is "noticetype" and 
- Option 2 : Put a star on fields signaled as "mandatory" where their only constraint is "noticetype" EXCEPT if they have a "forbidden" condition

**Example : For notice type 16 is it better if all fields in GR-Lot-PlannedDuration have a star OR none of them ? (BT-538-Lot is mandatory in fields.json but it is also forbidden under certain conditions**
![image](https://github.com/OP-TED/eForms-SDK/assets/158502029/c4c50fd4-c380-4670-808a-c924c73bccca)
![image](https://github.com/OP-TED/eForms-SDK/assets/158502029/0cde277b-036b-48f5-b028-bb4ca7510712)

**Comment by bertrand-lorentz** [comment, marked as answer]
The general logic is to first consider whether the field is forbidden, and if it is not, then consider whether it is mandatory.
So BT-538-Lot is forbidden if the condition is true (it makes no sense to indicate "another duration" if the duration is set in BT-36-Lot). And if it is not forbidden, then it is mandatory.

As shown in your screenshot, option 1 gives misleading information: the 4 fields cannot be filled in all together.
So option 2 is the way to go, with the following precision (you might have implied it): "EXCEPT if they have a "forbidden" condition **for the noticeType**"
